### PR TITLE
Validate nodeinfo response by schema

### PR DESCRIPTION
**Pull request by MFTabriz**
using a new matcher `match_json_schema` which uses json-schema gem. The schema file is also validated.

### CODE REVIEW

**Review by ineffyble — CHANGES_REQUESTED** [review]
Can you address the build failures? Looks like you need to regenerate the Gemfile.lock

**Comment by MFTabriz** [comment]
> Can you address the build failures? Looks like you need to regenerate the Gemfile.lock

I did a `bundle lock --update`. Now the tests pass for [ruby2.7](https://app.circleci.com/pipelines/github/mastodon/mastodon/67830/workflows/f0f3296b-2834-4885-b8be-5fd4d1b4e3dd/jobs/253460) but fail for [ruby3.0](https://app.circleci.com/pipelines/github/mastodon/mastodon/67830/workflows/f0f3296b-2834-4885-b8be-5fd4d1b4e3dd/jobs/253462). Did I break something? Can you please check this?

**Comment by MFTabriz** [comment]
Hi @ClearlyClaire ! Can you please check this out or ask someone to review it?

File: spec/support/matchers/json/match_json_schema.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+RSpec::Matchers.define :match_json_schema do |schema|
+  match do |input_json|
+    schema_directory = "#{Dir.pwd}/spec/support/schema/"
+    schema_path = "#{schema_directory}/#{schema}.json"
```
[the remark this answers is not in the document]
Comment (MFTabriz) [reply spec/support/matchers/json/match_json_schema.rb]: Done!

File: Gemfile.lock [hunk 2]
```diff
@@ -125,7 +125,7 @@ GEM
     bootsnap (1.15.0)
       msgpack (~> 1.2)
     brakeman (5.4.0)
-    browser (4.2.0)
+    browser (5.3.1)
```
Comment (ClearlyClaire) [inline Gemfile.lock:128]: Unrelated dependency update.

File: Gemfile.lock [hunk 3]
```diff
@@ -165,7 +165,7 @@ GEM
       activesupport
     cbor (0.5.9.6)
     charlock_holmes (0.7.7)
-    chewy (7.2.4)
+    chewy (7.2.7)
```
Comment (ClearlyClaire) [inline Gemfile.lock:168]: Unrelated dependency update.

File: Gemfile.lock [hunk 4]
```diff
@@ -481,7 +483,7 @@ GEM
       pry (>= 0.13, < 0.15)
     pry-rails (0.3.9)
       pry (>= 0.10.4)
-    public_suffix (5.0.0)
+    public_suffix (5.0.1)
```
Comment (ClearlyClaire) [inline Gemfile.lock:486]: Unrelated dependency update.

File: Gemfile.lock [hunk 5]
```diff
@@ -530,7 +532,7 @@ GEM
     rails-i18n (6.0.0)
       i18n (>= 0.7, < 2)
       railties (>= 6.0.0, < 7)
-    rails-settings-cached (0.6.6)
+    rails-settings-cached (0.7.2)
```
Comment (ClearlyClaire) [inline Gemfile.lock:535]: Unrelated dependency update.

File: Gemfile.lock [hunk 6]
```diff
@@ -545,7 +547,7 @@ GEM
     rdf-normalize (0.5.1)
       rdf (~> 3.2)
     redcarpet (3.5.1)
-    redis (4.5.1)
+    redis (4.8.0)
```
Comment (ClearlyClaire) [inline Gemfile.lock:550]: Unrelated dependency update.

File: Gemfile.lock [hunk 7]
```diff
@@ -585,7 +587,8 @@ GEM
     rspec-support (3.11.1)
     rspec_junit_formatter (0.6.0)
       rspec-core (>= 2, < 4, != 2.12.0)
-    rubocop (1.30.1)
+    rubocop (1.31.2)
+      json (~> 2.3)
```
Comment (ClearlyClaire) [inline Gemfile.lock:591]: Unrelated dependency update.

File: Gemfile.lock [hunk 8]
```diff
@@ -866,3 +870,9 @@ DEPENDENCIES
   webpacker (~> 5.4)
   webpush!
   xorcist (~> 1.1)
+
+RUBY VERSION
+   ruby 3.0.4p208
+
+BUNDLED WITH
+   2.2.33
```
Comment (ClearlyClaire) [inline Gemfile.lock:878]: This shouldn't be saved to the `Gemfile.lock`!

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
It looks like there's a bunch of unrelated dependency updates (I made in-line comments for some of them but not all). Can you revert those changes?

**Comment by MFTabriz** [comment]
> It looks like there's a bunch of unrelated dependency updates (I made in-line comments for some of them but not all). Can you revert those changes?

Thanks for checking it. I think I fixed all of them.

**Review by ClearlyClaire — APPROVED** [review]

**Comment by MFTabriz** [comment]
@ineffyble you asked for changez when this PR was a draft 3 weeks ago and you did not review it afterwards. I have to ask you again: Can you please check this out?

**Review by ineffyble — APPROVED** [review]

**Comment by ineffyble** [comment]
@MFTabriz Claire is the one you need sign-off from, I've just been trying to help with PR hygiene :)

**Pull request merged**